app/services/rebate_forms_service.rb

```diff
@@ -5,6 +5,28 @@ def initialize(rebate_form_attributes)
     @rebate_form_attributes = rebate_form_attributes
   end
 
+  def generate_qr(rebate_form)
+    # create a new token that only allows you to fetch a restricted set of details for this application only
+    # and use the same token to submit the signatures back
+    # this token is valid for 30 minutes
+
+    payload = {
+      rebate_form_id: rebate_form.id,
+      exp: Time.now.to_i + (1000 * 60),
+      per: 'fetch_application_and_submit_signatures'
+    }
+
+    token = JWT.encode payload, ENV['HMAC_SECRET'], 'HS256'
+
+    # iPad-application URL
+    url = ENV['APP_URL'] + 'admin/sign?jwt=' + token
```

Comment:
🐘 There are some railsy ways to write this which would make it more robust/idiomatic. Something like `url = admin_sign_url(jwt: token)`, see https://guides.rubyonrails.org/routing.html#path-and-url-helpers

If you do `rake routes` one of the columns is the corresponding helper function for the route, and they come in _path and _url variants

Is an APP_URL environment variable actually required? Rails usually knows where it's being served from, but that might not be true if there's a weird hosting setup.